### ISSUE TITLE
Stack Overflow Exception when reading a bad PDF

### ISSUE THREAD

**Issue by matthewpbowden** [jhabjan/Ghostscript.NET#17]
When we try to open an invalid PDF (we just took out the first few lines of a PDF file), we get a stack overflow exception because of the code here in the GhostscriptViewerPdfFormatHandler.cs:

```
                case PDF_PAGES_TAG:
                    {
                        string[] pages = rest.Split(new char[] { ' ' });
                        this.FirstPageNumber = int.Parse(pages[0]);
                        this.LastPageNumber = int.Parse(pages[1]);
                    }
```

To resolve we added this to the GhostscriptViewerStdIOHandler.cs:
                    string line = _outputMessages.ToString().Substring(0, rIndex);
                    if(line.ToUpper().Contains("UNRECOVERABLE"))
                    {
                        rIndex = -1;
                        return;
                    }

Basically, it keeps throwing string format exception when trying to parse the pages until the stack overlfows. Let me know if you need additional information. I'd be happy to add my fix to your code, but you may know of a better way to resolve the issue.

**Comment by jhabjan** [comment]
Hi, sorry for the delay. Can you please send me that pdf file so I can recreated the problem? I've tried to delete first couple of lines, but can't get it to output 'unrecoverable'..
